### DISCUSSION TITLE
Annotations list panel: Can we have explicit ordering?

### DISCUSSION THREAD

**Discussion by pkolyvas** [grafana/grafana#59770] [Dashboards, Panels & Visualizations]
## Summary 
Explicitly order the list of annotations on the Annotations List panel such that they're in order of newest to oldest? 

## Workflow
I like to use the Annotations List panel to surface annotations from across a variety of concerns so that I can see if an issue or change has been tagged/tracked/managed, etc. For my teams, their default dashboards include at least one Annotation panel to help surface points of interest. These Annotation Lists usually collect annotations from many other dashboards. 

## Problem
Some of my annotations lists are out of order. I can't figure out why. Using Grafana from source, with the Grafana data source and toy data, annotations seem to be well-ordered; they appear in a list from newest to oldest. Which is ideal. 

In my production instances, almost all the annotation lists are in reverse chronological order. This means the most current annotations are often off-screen, which isn't helpful. 

Example:
<img width="851" alt="Screenshot 2022-12-04 at 1 30 42 PM" src="https://user-images.githubusercontent.com/43178667/205508979-19380d14-0b28-4b21-b599-a66d9705b8dc.png">

## Suggested Solution
I think it might be easiest to simply check/ensure that the list of annotations returned is ordered by time and (perhaps) add a display option to flip between "sort by newest" / "sort by oldest".

I've got a working PR at the ready with this change if this is of interest (including a docs update): https://github.com/grafana/grafana/compare/main...pkolyvas:grafana:annolistOrder

It does the following:
- Ensures the ordering of the Annotations List is `newest first` no matter the order of the returned results
- Adds a `Display` option to toggle between `newest first` and `oldest first` (though I'm not even sure that's necessary; newest first is probably fine.)

## Other details of interest
- Grafana 9.2.5 & 9.3.1 both display the above issue
- I use Redis for caching with Grafana
- I use Postgres as the database

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
